packages/hardhat-core/src/internal/hardhat-network/provider/TxPool.ts

```diff
@@ -75,29 +75,79 @@ export class TxPool {
   constructor(
     private readonly _stateManager: PStateManager,
     blockGasLimit: BN,
+    minGasPrice: BN,
     common: Common
   ) {
     this._state = makePoolState({
       blockGasLimit: bnToHex(blockGasLimit),
+      minGasPrice: bnToHex(minGasPrice),
     });
     this._deserializeTransaction = (tx) => deserializeTransaction(tx, common);
   }
 
   public async addTransaction(tx: Transaction) {
     const senderAddress = this._getSenderAddress(tx);
-    const senderNonce = await this.getExecutableNonce(senderAddress);
+    const accountNonce = await this._getAccountNonce(senderAddress);
+    const nextNonce = await this.getNextNonce(senderAddress);
 
-    await this._validateTransaction(tx, senderAddress, senderNonce);
+    await this._validateTransaction(tx, senderAddress, accountNonce);
 
     const txNonce = new BN(tx.nonce);
 
-    if (txNonce.eq(senderNonce)) {
+    if (txNonce.lte(nextNonce)) {
```

Comment:
Any transaction that is `<= nextNonce` is considered a pending transaction. This assumes that `_validateTransaction` already checked that the tx nonce is not less than the account nonce.